### ISSUE TITLE
It needs to be more obvious that siteDir and url are mutually exclusive inputs

### ISSUE THREAD

**Issue by dbjorge** [microsoft/accessibility-insights-action#1068]
## Describe the bug

Several pilot teams onboarding to the Azure DevOps task have gotten confused about exactly which parameters they should be specifying when setting up the task. In particular, we've seen several pilot teams try to specify both `siteDir` and `url` inputs simultaneously, even though these parameters are mutually exclusive (specifying `url` causes `siteDir`, `localhostPort`, and `scanUrlRelativePath` to be ignored).

The task should do a better job of making it hard for a user to make this mistake. For users of ADO's graphical Pipeline definition UX (most of our pilot teams so far), I think the ideal would be to use task definition groups to enforce the parameters' mutually exclusivity as follows:

* Add a new required option that the user must select to determine whether they want to use "our task hosts your site for you using a simple Express configuration" mode vs "you host your site in a separate task before this one runs" mode
    * siteDir, scanUrlRelativePath, and localhostPort should be moved to a group with is only visible when "we host your site for you" mode is selected. In this mode, siteDir should be required and the others should be optional. The guidance for each should clarify the exact expected format (particularly, siteDir should clarify if it accepts relative vs absolute paths).
    * url should only be visible (and should be required) when "you host your site separately" mode is selected

However, this wouldn't do anything to help users of yaml build definitions. For them, I'd suggest the following additions:

* add an early parameter validation step before starting to scan which detects mutually exclusive parameters being set and fails fast with an actionable error message, rather than silently picking one set to ignore
* Change the names of the inputs to make the grouping between them more clear and to make it easier for documentation to refer to the parameters as a group (eg, for error messages and the `url` documentation to explain what it's mutually exclusive with). Perhaps something like:
    * `staticSiteDir` (instead of `siteDir`)
    * `staticSitePort` (instead of `localhostPort`)
    * `staticSiteUrlRelativePath` (instead of `scanUrlRelativePath`)

Note that most of the above suggestions would be breaking changes; see #1063 

**To Reproduce**

1. create a test pipeline using the ADO task which specifies both `siteDir` and `url`
2. run a build
3. observe that the task silently ignores `siteDir`

## CodePen repro example

n/a

## Expected behavior

We should strive for our UX to be a "pit of success" - we should make it hard for users to misconfigure the product, especially in a way which misbehaves silently. It isn't enough to rely on users reading usage documentation.

## Screenshots

n/a

## Context (please complete the following information)

- OS Name & Version: n/a
- AI-Action Version & Environment: v1.1.1 production
- Browser Version: n/a
- Target Page: n/a

## Are you willing to submit a PR?

Yes

## Did you search for similar existing issues?

yes

## Additional context

n/a
